neqo-transport/src/recovery.rs

```diff
@@ -209,12 +227,152 @@ impl LossRecoverySpaces {
     }
 }
 
+#[derive(Debug)]
+struct CongestionControl {
+    congestion_window: u64, // = kInitialWindow
+    bytes_in_flight: u64,
+    congestion_recovery_start_time: Option<Instant>,
+    ssthresh: u64,
+}
+
+impl Default for CongestionControl {
+    fn default() -> Self {
+        CongestionControl {
+            congestion_window: INITIAL_WINDOW,
+            bytes_in_flight: 0,
+            congestion_recovery_start_time: None,
+            ssthresh: u64::max_value(),
+        }
+    }
+}
+
+impl Display for CongestionControl {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "CongCtrl")
+    }
+}
+
+impl CongestionControl {
+    #[cfg(test)]
+    pub fn cwnd(&self) -> u64 {
+        self.congestion_window
+    }
+
+    fn cwnd_avail(&self) -> u64 {
+        // BIF can be higher than cwnd due to PTO packets, which are sent even
+        // if avail is 0, but still count towards BIF.
+        self.congestion_window.saturating_sub(self.bytes_in_flight)
+    }
+
+    // OnPacketAckedCC
+    fn on_packet_acked(&mut self, pkt: &SentPacket) {
+        assert!(self.bytes_in_flight >= u64::try_from(pkt.size).unwrap());
+        self.bytes_in_flight -= u64::try_from(pkt.size).unwrap();
+
+        if self.in_congestion_recovery(pkt.time_sent) {
+            // Do not increase congestion window in recovery period.
+            return;
+        }
+        if self.app_limited() {
+            // Do not increase congestion_window if application limited.
+            return;
+        }
+
+        if self.congestion_window < self.ssthresh {
+            // Slow start.
+            self.congestion_window += u64::try_from(pkt.size).unwrap();
+            qinfo!([self], "slow start; cwnd {}", self.congestion_window);
+        } else {
+            // Congestion avoidance.
+            qinfo!(
+                [self],
+                "congestion avoidance; cwnd {} ssthresh {}",
+                self.congestion_window,
+                self.ssthresh
+            );
+            self.congestion_window += u64::try_from(MAX_DATAGRAM_SIZE * pkt.size as usize).unwrap()
```

Comment:
We should just use the current pmtu here.  Have a setter for it, probably.

Reply:
This is broken for the same reason as #349 -- Congestion Control currently doesn't know the PMTU. I think we can tackle this in a later PR, using MAX_DATAGRAM_SIZE should still work for the moment.